vm/package/deploy.rs

```diff
@@ -14,31 +14,38 @@
 // You should have received a copy of the GNU General Public License
 // along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
 
-use snarkvm_compiler::Deployment;
-use snarkvm_console::types::Address;
+use snarkvm_compiler::Transaction;
+use snarkvm_console::{
+    program::{Plaintext, Record},
+    types::Address,
+};
 
 use super::*;
 
 pub struct DeployRequest<N: Network> {
-    deployment: Deployment<N>,
+    transaction: Transaction<N>,
     address: Address<N>,
     program_id: ProgramID<N>,
 }
 
 impl<N: Network> DeployRequest<N> {
     /// Sends the request to the given endpoint.
-    pub fn new(deployment: Deployment<N>, address: Address<N>, program_id: ProgramID<N>) -> Self {
-        Self { deployment, address, program_id }
+    pub fn new(transaction: Transaction<N>, address: Address<N>, program_id: ProgramID<N>) -> Result<Self> {
+        ensure!(
+            matches!(transaction, Transaction::Deploy(_, _, _)),
+            "Cannot create a deploy request with an execution transaction"
```

Comment:
```suggestion
            "Cannot create a deploy request with a non-deployment transaction"
```

Reply:
^We intend to support other transaction types in the future, hence the greater specificity.